### DISCUSSION TITLE
Welcome to deep-translator Discussions!

### DISCUSSION THREAD

**Discussion by nidhaloff** [nidhaloff/deep-translator#59] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by emm369** [comment]
hello, i have a question, which might sound weird, but I have to make sure : 
is the google translator used in this API is the official API of google ? i mean, is it the restricted and the official free version, or.. ? in other words, is it legal to use this package including google translate ?

**Reply by nidhaloff** [reply]
@emm369 No the question is not weird, I can understand why are you questioning this. 

Now to answer it, deep-translator does not use the Google API at all. It uses web scraping in the background to scrap the whole HTML body and then extract the translation from it. Therefore, the package does not rely on the Google API, which makes it hustle-free to use because other API-based packages do not work anymore or they tend to be buggy and not reliable. That's why this package was created in the first place.

Now for the second part of your question, which is whether it is legal to use it. Well, there are no issues with web scraping. It is basically like visiting a website, right-click, and then clicking on view source, which will show you the HTML body of the site. You can do this with any site so, of course, it's legal otherwise you wouldn't/shouldn't have access to the site at all. 

A solution against web scraping is for example dynamic page loading, which means the HTML will not be loaded when a visitor access the website. Many websites use this to prevent people (or bots) from scraping them. Well, Google could have implemented this if they wanted to, so as long as they didn't do it, deep-translator should work fine. However, there are also other workarounds to make this work even if the website implements dynamic HTML loading, but in this case, I would worry if it's legal. So I wouldn't go further and implement that in deep-translator if google would prevent web scraping in the future. 

Hope I answered your question.

**Comment by emm369** [comment]
after using your API for a while i kinda got confused, 
well, it's clear to me that google translate is not blocking scrapers(although i think it's being minimalized) and regarding the fact that they have their own API, is still legal to scrape from them ? ( be it translations or audio) - i read the terms of use, but i am note sure
and by the way, i appreciate your explanation

**Reply by nidhaloff** [reply]
@emm369 So, did you find something against scraping in their terms of use?

**Reply by emm369** [reply]
as far as i read no, but i couldn't be sure as am no specialist, i might have missed something, people are saying, if google has its own API, paid after a certain limit then scraping shouldn't be used or something, am not sure if i got it right

**Reply by nidhaloff** [reply]
@emm369 I'm certainly no specialist too. Like I said scraping, in general, is totally fine and people/developers have been doing this all the time. There are a lot of frameworks for scraping a website to search for some information and deep-translator is doing just that.
